packages/ui/tsup.config.ts

```diff
@@ -1,14 +1,20 @@
-import { sassPlugin, postcssModules } from 'esbuild-sass-plugin';
+import { postcssModules, sassPlugin } from 'esbuild-sass-plugin';
 import { defineConfig } from 'tsup';
+import pkg from './package.json' assert { type: 'json' };
+
+const entry = Object.values( pkg.exports ).map( ( entry ) => entry[ 'calypso:src' ] );
 
 export default defineConfig( {
-	entry: [ 'src/index.ts' ],
+	entry,
 	clean: true,
 	splitting: true,
 	experimentalDts: true,
 	sourcemap: true,
 	format: [ 'esm', 'cjs' ],
 	outDir: 'dist',
+	banner: {
+		js: "\nimport './index.css';\n",
```

Comment:
To really lean into code splitting, I wonder if we'd be better off to require not just CSS imports, but CSS modules imports, and then we avoid compiling a single CSS output like this and leave the original CSS module import statements to be resolved in the component source.